tests/test_deduplication_performance.py

```
@@ -0,0 +1,375 @@
+    def test_memory_usage_deterministic_ids(self):
+        """Test memory usage of deterministic ID generation."""
+        rag_service = RagService()
+
+        # Generate many deterministic IDs and measure memory efficiency
+        num_urls = 1000
+        urls = [f"https://example.com/page-{i}" for i in range(num_urls)]
+
+        # Test ID generation performance
+        start_time = time.time()
+        ids = []
+        for i, url in enumerate(urls):
```

Ruff (0.12.2)

Loop control variable `i` not used within loop body

(B007)